tests/storage/test_unsafe_locale.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest.mock import patch
+
+from synapse.storage.database import make_conn
+from synapse.storage.engines._base import IncorrectDatabaseSetup
+from synapse.storage.engines.postgres import PostgresEngine
+
+from tests.unittest import HomeserverTestCase
+
+
+class UnsafeLocaleTest(HomeserverTestCase):
+    @patch("synapse.storage.engines.postgres.PostgresEngine.get_db_locale")
+    def test_unsafe_locale(self, mock_db_locale):
```

Comment:
```suggestion
    def test_unsafe_locale(self, mock_db_locale) -> None:
```
Just so we can claim this file is type-checked :innocent: :-) (occasionally tests can be good places that exercise the type annotations a bit, providing valid type errors for mistakes in the actual code. Not sure it'd buy much here, but why not)

Don't know what the type of the Mock is, would be tempted to put it in but also suppose it stands very little chance of mattering.